sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -499,7 +499,35 @@ static size_t _encode_uint(uint32_t *val)
     return size;
 }
 
-static unsigned _put_delta_optlen(uint8_t *buf, unsigned offset, unsigned shift, unsigned val)
+/*
+ * Writes CoAP Option header. Expected to be called twice to write an option:
```

Comment:
nice documentation, 

I would suggest (even though .c files are not included with Doxygen) you might want to use proper tags here, i.e.,

- `@brief` at the beginning 
- `@param` and
- `@return`

for the latter two you missed to use the `@`, the advantage is that someone could still run doxygen including .c files to generate a nice docu.

Reply:
ah well, looking through the rest of the code its never used or done that way, and its an internal function, so its seems not that important.